config/v1/types_cluster_version.go

```diff
@@ -691,6 +691,15 @@ type Update struct {
 // Release represents an OpenShift release image and associated metadata.
 // +k8s:deepcopy-gen=true
 type Release struct {
+	// architecture is an optional field that indicates the
+	// value of the cluster architecture. In this context cluster
```

Comment:
can we also clarify in the comment that this is the architecture the cluster is trying to reconcile to and not necessarily the current payload architecture

Reply:
Not at this level, because `Release` shows up in other places too:

```console
$ git grep Release config/ | grep json
config/v1/types_cluster_version.go:     Desired Release `json:"desired"`
config/v1/types_cluster_version.go:     AvailableUpdates []Release `json:"availableUpdates"`
config/v1/types_cluster_version.go:     Release Release `json:"release"`
```

But up one level above this, at the `status.desired` consumer you're thinking of, we [already have][1]:

> desired is the version that the cluster is reconciling towards

which addresses your concern, right?  Or did you want additional wording there to underline "but 'reconciling towards' does not mean 'completely delivered'"?

[1]: https://github.com/openshift/api/blob/3e5de946111c67cb1f2b09ff7da569012d15931d/config/v1/types_cluster_version.go#L138

Reply:
I think we are fine as is. As you said, given that the wording is there in the `desired` field, we are fine.